src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestService.swift

```diff
@@ -674,7 +674,16 @@ class CoronaTestService {
 					}
 
 					// only store test result in diary if negative or positive
-					if (testResult == .positive || testResult == .negative) && !coronaTest.journalEntryCreated {
+					// Warning: check the current coronaTest so that changes are not overlooked
+					//
+					if let journalEntryCreated = self.coronaTest(ofType: coronaTestType)?.journalEntryCreated,
+					   (testResult == .positive || testResult == .negative) && !journalEntryCreated {
+						switch coronaTestType {
+						case .pcr:
+							self.pcrTest?.journalEntryCreated = true
+						case .antigen:
+							self.antigenTest?.journalEntryCreated = true
+						}
```

Comment:
why does this block has to be put above the `addCoronaTest` call?

Reply:
the add coronTest call will always succeed, to make sure it won't gat called twice and give the sql database some time